### PR TITLE
Tree Icon Improvements

### PR DESCRIPTION
**Pull request by amcdnl**
This PR:

- Makes it possible to pass in custom icons for tree
- Abstracts Arrow into its own component
- Adds an Arrow story and CustomArrow story to Tree

image

### CODE REVIEW

File: src/layout/Tree/TreeContext.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export interface TreeContextProps {
+  /**
+   * A custom icon to be used for collapsed nodes.
+   */
+  collapsedIcon?: any;
+
+  /**
+   * A custom icon to be used for expanded nodes.
+   */
+  expandedIcon?: any;
```
Comment (shaswat-indian) [inline src/layout/Tree/TreeContext.ts:12]: Can the icon types be more specific?

**Review by shaswat-indian — COMMENTED** [review]
Looks good, just have a question around the icon prop types.

I'm also thinking if we should add some more flexibility to the Tree like:-
- Different Expand/Collapse icons for each level of the nested tree. The `collapsedIcon` and `expandedIcon` props could be a list passed to the Tree component in that case, each element corresponding to the respective index (level of the tree).
- Flexibility to have custom Expand/Collapse icons for the TreeNode as well, just in case there are some special nodes which need to be highlighted.

File: src/layout/Tree/TreeContext.ts (hunk 1 above)
Comment (amcdnl) [reply src/layout/Tree/TreeContext.ts:12]: Agreed. We need to audit the types all over anyways.

File: src/elements/Arrow/Arrow.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  /**
+   * CSS Classname to apply to the arrow
+   */
+  className?: string;
+
+  /**
+   * The direction of the arrow
+   */
+  direction?: 'up' | 'right' | 'down' | 'left';
+}
```
Comment (steppy452) [inline src/elements/Arrow/Arrow.tsx:15]: @amcdnl was thinking about this.. I wonder if we should do a `size` prop here and pass it into the width/height in the svg .. and then maybe use `currentColor` as the fill by default which can get overridden with a className. I didn't quite like how I added an extra className to the default icons for the Tree component

Comment (amcdnl) [reply src/elements/Arrow/Arrow.tsx:15]: It should have inherited it from the CSS in the button.

**Pull request merged**